feat: remove vaults from vault factory

eliseoci opened the pull request:
## Description and Motivation

<!-- 
    
    Please write a description of what this PR is changing, removing or adding, and why. Consider including before/after 
    comparisons.

-->

This PR adds a message on the vault factory contract to remove vaults added to the vault directory. This way, in case wrong vaults are added, they can be removed and keep the vault directory clean.
Implemented `ExecuteMsg::RemoveVault` in a slightly different way than requested in [Issue #52](https://github.com/White-Whale-Defi-Platform/migaloo-core/issues/52):
I decided to use AssetInfo instead of the vault address as vault identifier because AssetInfo is the key for the Vaults Map.

## Related Issues

<!-- 
    
    Add the list of issues related to this PR from the [issue tracker](https://github.com/White-Whale-Defi-Platform/migaloo-core/issues).
    Indicate, which of these issues are resolved or fixed by this PR, like #XXXX, where XXXX is the issue number.

-->
#52 

---
## Checklist:

<!-- 

    Thanks for contributing to White Whale Migaloo! 
    
    Before you file this pull request, please follow the items on this checklist and put an x in each of the boxes, 
    like this: [x]. 
    
    Make sure to follow the guidelines, so we can process this PR as fast as possible. 

-->

- [x] I have read [Migaloo's contribution guidelines](https://github.com/White-Whale-Defi-Platform/migaloo-core/blob/main/CONTRIBUTING.md).
- [x] My pull request has a sound title and description (not something vague like `Update index.md`)
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation.
- [x] The code is formatted properly `cargo fmt --all --`.
- [x] Clippy doesn't report any issues `cargo clippy -- -D warnings`.
- [x] I have regenerated the schemas if needed `cargo schema`.

kerber0x reviewed (changes requested):
Hello @eliseoci and thanks for contributing to migaloo-core! Your PR looks good, only one thing missing, a test where you try to remove a vault that doesn't exist you should get the `VaultFactoryError::NonExistentVault`. Get that done and we'll merge it :star2:

Also, make sure to reword the commit to follow the [conventional commits syntax](https://www.conventionalcommits.org/en/v1.0.0/#summary).

eliseoci commented:
> Hello @eliseoci and thanks for contributing to migaloo-core! Your PR looks good, only one thing missing, a test where you try to remove a vault that doesn't exist you should get the `VaultFactoryError::NonExistentVault`. Get that done and we'll merge it star2
> 
> Also, make sure to reword the commit to follow the [conventional commits syntax](https://www.conventionalcommits.org/en/v1.0.0/#summary).

Hi @kerber0x, thanks for the feedback. I've addressed your comments.
